### PR TITLE
Add new notifications popup

### PR DESCRIPTION
**Pull request by AdityaKhatri**
## Changes

* Add new notifications popup
Depends on: https://github.com/the-deep/server/pull/1022

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: app/components/Notifications/NotificationContainer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+import {
+    NotificationStatusUpdateMutation,
+    NotificationStatusUpdateMutationVariables,
+} from '#generated/types';
+
+import { Notification } from '../types';
+import styles from './styles.css';
+
+const NOTIFICATION_STATUS_UPDATE = gql`
+mutation NotificationStatusUpdate($notificationId: String!, $newStatus: status!) {
```
Comment (samshara) [inline app/components/Notifications/NotificationContainer/index.tsx]: update type with latest server value

File: app/components/Notifications/NotificationContainer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    ] = useModalState(false);
+
+    const alert = useAlert();
+
+    const [
+        updateStatus,
+    ] = useMutation<NotificationStatusUpdateMutation, NotificationStatusUpdateMutationVariables>(
+        NOTIFICATION_STATUS_UPDATE,
+        {
+            onCompleted: (response) => {
```
Comment (samshara) [inline app/components/Notifications/NotificationContainer/index.tsx:89]: shouldn't we re-fetch notifications after update?

File: app/components/Notifications/NotificationContainer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                if (response?.notificationStatusUpdate?.ok) {
+                    alert.show(
+                        'Successfully changed url status.',
```
Comment (samshara) [inline app/components/Notifications/NotificationContainer/index.tsx]: is `url status` approprite name?

File: app/components/Notifications/NotificationContainer/index.tsx (hunk 2 above)
Comment (AdityaKhatri) [reply app/components/Notifications/NotificationContainer/index.tsx:89]: This happens automatically with the help of apollo.

**Review by samshara — APPROVED** [review]

**Pull request merged**